signer/keydbstore/couch_keydbstore.go

```diff
@@ -0,0 +1,434 @@
+}
+
+// GetKey returns the PublicKey given a KeyID, and does not activate the key
+func (cdb *CouchDBKeyStore) GetKey(keyID string) data.PublicKey {
+	dbPrivateKey, _, err := cdb.getKey(keyID)
+	if err != nil {
+		return nil
+	}
+
+	return data.NewPublicKey(dbPrivateKey.Algorithm, dbPrivateKey.Public)
+}
+
+// ListKeys always returns nil. This method is here to satisfy the CryptoService interface
+func (cdb CouchDBKeyStore) ListKeys(role data.RoleName) []string {
+	return nil
+}
+
+// ListAllKeys always returns nil. This method is here to satisfy the CryptoService interface
+func (cdb CouchDBKeyStore) ListAllKeys() map[string]data.RoleName {
+	return nil
+}
+
+// RemoveKey removes the key from the table
+func (cdb CouchDBKeyStore) RemoveKey(keyID string) error {
+	// Delete the key from the database
+	dbPrivateKey := CDBPrivateKey{KeyID: keyID}
+
+	db, err := couchdb.DB(cdb.client, cdb.dbName, dbPrivateKey.TableName())
+	if err != nil {
+		return err
+	}
+
+	rows, err := db.Find(context.TODO(), map[string]interface{}{
+		"selector": map[string]interface{}{
+			"key_id": map[string]string{
+				"$eq": keyID,
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		if err := rows.ScanDoc(&dbPrivateKey); err != nil {
+			return fmt.Errorf("Could not ScanDoc result: %s", err)
+		}
+		if _, err = db.Delete(context.TODO(), keyID, dbPrivateKey.ID.Rev); err != nil {
+			return fmt.Errorf("unable to delete private key %s from database: %s", keyID, err.Error())
+		}
+	}
+
+	return nil
+}
+
+// RotateKeyPassphrase rotates the key-encryption-key
+func (cdb CouchDBKeyStore) RotateKeyPassphrase(keyID, newPassphraseAlias string) error {
+	dbPrivateKey, decryptedPrivKey, err := cdb.getKey(keyID)
+	if err != nil {
+		return err
+	}
+
+	// Get the new passphrase to use for this key
+	newPassphrase, _, err := cdb.retriever(dbPrivateKey.KeyID, newPassphraseAlias, false, 1)
+	if err != nil {
+		return err
+	}
+
+	// Re-encrypt the private bytes with the new passphrase
+	newEncryptedKey, err := jose.Encrypt(decryptedPrivKey, KeywrapAlg, EncryptionAlg, newPassphrase)
+	if err != nil {
+		return err
+	}
+
+	// Update the database object
+	dbPrivateKey.Private = []byte(newEncryptedKey)
+	dbPrivateKey.PassphraseAlias = newPassphraseAlias
+
+	db, err := couchdb.DB(cdb.client, cdb.dbName, dbPrivateKey.TableName())
+	if err != nil {
+		return err
+	}
+
+	rows, err := db.Find(context.TODO(), map[string]interface{}{
+		"selector": map[string]interface{}{
+			"key_id": map[string]string{
+				"$eq": keyID,
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var doc CDBPrivateKey
+		rows.ScanDoc(&doc)
+		doc.Private = []byte(newEncryptedKey)
+		doc.PassphraseAlias = newPassphraseAlias
+		doc.ID.ID = keyID
+		if _, err := db.Put(context.TODO(), keyID, doc); err != nil {
+			return err
+		}
```

Comment:
Non-blocking:  there should only be 1 key with this key ID I think - should this return after?